### PR TITLE
Revamp ITR documentation

### PR DESCRIPTION
**Pull request by edrevo**
### What does this PR do?
Revamp the ITR documentation:
- Include more content in the landing page for ITR, removing things that do not apply anymore and highlighting what the product does.
- Split out language setup to their own pages
- Add link to ITR from Test Visibility setup 
- Minimized the instructions for ITR setup to include only the ITR-specific stuff. For example, if users want to user agentless or the agent there's already better documentation about that in the test visibility setup.
- Fixed some command line bugs in the JS instructions.

### Motivation
@andrea-mosk told me our current doc was a bit depressing. I agree.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/intelligent_test_runner/javascript.md
Comment (edrevo) [inline content/en/continuous_integration/intelligent_test_runner/javascript.md:20]: Need confirmation from @juan-fernandez about these

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (edrevo) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:16]: The text in this page I've basically copied from https://www.datadoghq.com/blog/streamline-ci-testing-with-datadog-intelligent-test-runner/

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (edrevo) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:80]: Need to confirm this link with @drodriguezhdez

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (andrea-mosk) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:14]:
```suggestion
<div class="alert alert-warning">Intelligent Test Runner is a beta product.</div>
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (KickingTheTV) [reply content/en/continuous_integration/intelligent_test_runner/_index.md:16]: "Test Impact Analysis" is the industry term for what we're doing, the analysis of which tests are impacted. I think it's helpful to include in terms of SEO (Search engine optimization). Datadog docs pages rank quite high with google, so hopefully when someone searches for "Test impact analysis" our docs page can show up.

I also agree that "testing downtime" is unclear/misleading.

What do you think about this combination of suggestions:

"Intelligent Test Runner is Datadog's test impact analysis solution. It automatically and intelligently selects and runs only the relevant tests for a given commit based on the code being changed. Significantly reduce time spent testing and overall CI costs, while maintaining test coverage."

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (KickingTheTV) [reply content/en/continuous_integration/intelligent_test_runner/_index.md]: I would change this last sentence to something more like:

"...so you can understand what parts of your organization are using and getting the most out of Intelligent Test Runner."

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:16]: "automatically and intelligently" sounds more like marking material than docs. That's just my opinion though

Comment (juan-fernandez) [reply content/en/continuous_integration/intelligent_test_runner/_index.md:16]: I'm sure the docs team will come up with something better but I'd rather do something like

> It uses the code changed in a given commit to automatically select and run only the relevant tests.

```suggestion
Intelligent Test Runner is Datadog's test impact analysis solution. It uses the code changed in a given commit to automatically select and run only the relevant tests. It significantly reduces time spent testing and overall CI costs, while maintaining test coverage.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:25]: is that a limitation only during the beta? We might want to set expectations here

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:47]: the last phrase does not explain much, does it? 
Maybe 
```suggestion
Due to the limitations described above, the default branch of your repository is automatically excluded from having Intelligent Test Runner enabled. This is the recommended configuration by Datadog to minimize the risks of false positives. 
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:49]: maybe add an example? 
```suggestion
If there are other branches you want to exclude, you can add them from the Intelligent Test Runner settings page. The query bar supports the wildcard character `*` to exclude any branches that match, such as `release_*`
```

File: content/en/continuous_integration/intelligent_test_runner/javascript.md
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/javascript.md:20]: let's set them to these versions, as they include a bug fix
```suggestion
  * From `dd-trace>=3.11.0` or `dd-trace>=2.24.0`
```

File: content/en/continuous_integration/intelligent_test_runner/javascript.md
Comment (juan-fernandez) [inline content/en/continuous_integration/intelligent_test_runner/javascript.md:24]:
```suggestion
  * From `dd-trace>=3.11.0` or `dd-trace>=2.24.0`
```

**Review by juan-fernandez — APPROVED** [review]
left some minor comments but I like the revamp. Thanks for this!

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (edrevo) [reply content/en/continuous_integration/intelligent_test_runner/_index.md:16]: I see this initial text is contentious! Haha! There's already a thread about this in https://github.com/DataDog/documentation/pull/17286#discussion_r1135944042

@KickingTheTV, I think you have the last word here.

**Review by andrea-mosk — APPROVED** [review]
Thanks a lot for the effort!

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (edrevo) [reply content/en/continuous_integration/intelligent_test_runner/_index.md]: Discussed offline and rephrased.

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:31]:
```suggestion
## Set up Datadog library
```
Let's take the opportunity to bring this heading into compliance with our capitalization and wording standards.

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:33]:
```suggestion
Prior to setting up Intelligent Test Runner, you must set up [Test Visibility][1] for your particular language. If you are reporting data through the Agent, use v6.40+/v7.40+.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:43]:
```suggestion
Once you have setup your Datadog library for Intelligent Test Runner, configure it from the [Test Service Settings][2] page.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:47]:
```suggestion
Due to the limitations described above, the default branch of your repository is automatically excluded from having Intelligent Test Runner enabled. Datadog recommends this configuration to ensure that all of your tests run prior to reaching production.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:49]:
```suggestion
If there are other branches you want to exclude, add them from the Intelligent Test Runner settings page. The query bar supports using the wildcard character `*` to exclude any branches that match, such as `release_*`
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:55]:
```suggestion
You can explore the time savings you get from Intelligent Test Runner by looking at the test commit page and test sessions panel.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:61]:
```suggestion
When Intelligent Test Runner is active and skipping tests, purple text displays the amount of time saved on each test session or on each commit. The duration bar also changes color to purple so you can quickly identify which test sessions are using Intelligent Test Runner in the [Test Runs][3] page.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:65]:
```suggestion
Track you organization's savings and adoption of Intelligent Test Runner through the out-of-the-box [Intelligent Test Runner dashboard][4]. The dashboard includes widgets to track your overall savings as well as a per-repository, per-committer, and per-service view of the data. View the dashboard to understand which parts of your organization are using and getting the most out of Intelligent Test Runner.
```

File: content/en/continuous_integration/intelligent_test_runner/_index.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/_index.md:69]:
```suggestion
The dashboard also tracks adoption of Intelligent Test Runner throughout your organization.
```

File: content/en/continuous_integration/intelligent_test_runner/dotnet.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/dotnet.md:21]:
```suggestion
Prior to setting up Intelligent Test Runner, set up [Test Visibility for .NET][1]. If you are reporting data through the Agent, use v6.40+/v7.40+.
```

File: content/en/continuous_integration/intelligent_test_runner/dotnet.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/dotnet.md:23]:
```suggestion
To enable Intelligent Test Runner, set the following environment variables:
```

File: content/en/continuous_integration/intelligent_test_runner/javascript.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/javascript.md:28]:
```suggestion
Prior to setting up Intelligent Test Runner, set up [Test Visibility for Javascript and Typescript][1]. If you are reporting data through the Agent, use v6.40+/v7.40+.
```

File: content/en/continuous_integration/intelligent_test_runner/javascript.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/javascript.md:30]:
```suggestion
To enable Intelligent Test Runner, set the following environment variables:
```

File: content/en/continuous_integration/intelligent_test_runner/swift.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/swift.md:21]:
```suggestion
Prior to setting up Intelligent Test Runner, set up [Test Visibility for Swift][1]. The **code coverage** option must also be enabled in the test settings of your scheme or test plan, or --enable-code-coverage must be added to your Swift test command (if using a SPM target).
```

File: content/en/continuous_integration/intelligent_test_runner/swift.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/swift.md:23]:
```suggestion
If you are reporting data through the Agent, use v6.40+/v7.40+.
```

File: content/en/continuous_integration/intelligent_test_runner/swift.md
Comment (urseberry) [inline content/en/continuous_integration/intelligent_test_runner/swift.md:25]:
```suggestion
To enable Intelligent Test Runner, set the following environment variables:
```

**Comment by edrevo** [comment]
Many thanks for the review @urseberry! Just FYI, there are 10 instances of `Datadog Library` (with that capitalization) throughout the documentation (not just this one, but all of Datadog's docs), in case you want to open a PR to normalize that.

**Review by urseberry — APPROVED** [review]
Found one more small nit. Approving so I don't hold up the PR.

**Comment by edrevo** [comment]
`dd-gitlab/set_redirect_metadata_preview` seems to have failed but there's no link to gitlab to check what went wrong. do you know what could have cause it to fail @urseberry?

**Comment by urseberry** [comment]
> `dd-gitlab/set_redirect_metadata_preview` seems to have failed but there's no link to gitlab to check what went wrong. do you know what could have cause it to fail @urseberry?

Here is the link: https://gitlab.ddbuild.io/DataDog/documentation/-/jobs/241224642
The job is related to some work the Websites team is doing in the documentation repo. It is unrelated to your PR.

I'll go ahead and merge this. Thanks again!

**Pull request merged**
